### PR TITLE
fix: Changes email validation to match @LearningLocker.

### PR DESCRIPTION
**Pull request by AndyHubert**
**Description**
- Changes email validation to use the same regex as Learning Locker.

**Related Issues**
- Follows on from #229

**PR Type**
- Fix

### CODE REVIEW

File: src/transformer/utils/get_user.php [hunk 1]
```diff
@@ -19,7 +19,8 @@
 
 function get_user(array $config, \stdClass $user) {
     $fullname = get_full_name($user);
-    $hasvalidemail = filter_var($user->email, FILTER_VALIDATE_EMAIL);
+    // The following email validation matches that in Learning Locker
+    $hasvalidemail = mb_ereg_match("[A-Z0-9\\.\\`\\'_%+-]+@[A-Z0-9.-]+\\.[A-Z]{1,63}$", $user->email, "i");
```
Comment (ryasmi) [inline src/transformer/utils/get_user.php:23]: The regex here will need a `^` at the beginning to ensure that it matches from the beginning of the string.

**Review by ryasmi — CHANGES_REQUESTED** [review]

**Comment by ryasmi** [comment]
Thanks @AndyHubert, looks good besides that one small point in the regex.

File: src/transformer/utils/get_user.php (hunk 1 above)
Comment (AndyHubert) [reply src/transformer/utils/get_user.php:23]: The [mb_ereg_match](http://php.net/manual/en/function.mb-ereg-match.php) function matches from the beginning of the string automatically.

Comment (ryasmi) [reply src/transformer/utils/get_user.php:23]: Ah didn't know that, thanks @AndyHubert, seems good then 👍

**Review by ryasmi — APPROVED** [review]

**Pull request merged**